### PR TITLE
Update override functions to set ENV vars

### PR DESCRIPTION
**Pull request by alexstoick**
Using the ENV variables to reliably determine what branch / version we are running on.

### CODE REVIEW

File: cli/pipeline/pipeline_definition_test.go [hunk 1]
```diff
@@ -48,13 +48,16 @@ func TestOverrideVersion(t *testing.T) {
 	}
 	pipeline := parsePipeline(data)
 
-	pipeline.Steps[0].OverrideVersion("", true)
+  var flags = runCmdFlagsStruct{}
+  flags.OverrideInputs = true
```
Comment (knaveofdiamonds) [inline cli/pipeline/pipeline_definition_test.go]: Indentation?

Comment (alexstoick) [reply cli/pipeline/pipeline_definition_test.go]: ffs gofmt.

**Comment by alexstoick** [comment]
@knaveofdiamonds Since we can no longer reliably determine the branch from `pipeline-steps.yml` because of the overrides that are passed into `paddle` I'm setting two env variables *step_version* and *step_branch* that allow me to determine if the branch has been overwritten.

**Comment by me** [comment]
I agree it's a bit difficult to follow... What's the use case? Could we instead set `step_version` and `step_branch` manually in the env if that's needed?

**Comment by alexstoick** [comment]
guess so - just have to remember to do that when we override the branch :+1:

**Pull request closed without merging**